modules/@apostrophecms/util/ui/public/2-http.js

```diff
@@ -127,7 +129,7 @@
         } else {
           query = {};
         }
-        query['apos-mode'] = 'draft';
+        query['apos-mode'] = options.draft ? 'draft' : 'published';
```

Comment:
You need an `undefined` check on `options.draft` here, otherwise something is passed all the time, which is not always appropriate.

Reply:
@boutell isn't [this check](https://github.com/apostrophecms/apostrophe/pull/2956/files#diff-de132b0a468c668b0f24c2aaa7cba4687c855728c2897417b445377ef9f9bcd4R117) going to safely skip this block in the case of `options.draft` being `undefined` ?

https://github.com/apostrophecms/apostrophe/pull/2956/files#diff-de132b0a468c668b0f24c2aaa7cba4687c855728c2897417b445377ef9f9bcd4R117

Reply:
There are two forks of this if statement, the one for options.qs being truthy is correct, but the one for options.qs not being truthy contains the bad assumption that options.draft should always do something (line 132).

Reply:
@boutell is this check enough https://github.com/apostrophecms/apostrophe/pull/2956/files#diff-de132b0a468c668b0f24c2aaa7cba4687c855728c2897417b445377ef9f9bcd4R132-R134 ?